src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -684,6 +717,63 @@ private static bool InitRTNoCacheSupported()
                 "NoCache");
         }
 
+        private void RTServerCertificateCallback(RTHttpBaseProtocolFilter sender, RTHttpServerCustomValidationRequestedEventArgs args)
+        {
+            // Convert WinRT certificate to .NET certificate.
+            X509Certificate2 serverCert = ConvertCertificate(args.ServerCertificate);
+
+            // Create .NET X509Chain from the WinRT information. We need to rebuild
+            // the chain since WinRT only gives us an array of intermediate certificates.
+            var serverChain = new X509Chain();
```

Comment:
I checked .NET Framework.  It does indeed dispose of the chain and server certificate after the callback is done.  So, I will revise this code to do the same.

For reference, here is a summary of the .NET Framework code:
https://github.com/Microsoft/referencesource/blob/master/System/net/System/Net/_SecureChannel.cs#L1281

```c#
if (remoteCertValidationCallback != null)
{
    success = remoteCertValidationCallback(m_HostName, remoteCertificateEx, chain, sslPolicyErrors);
}

// ...
finally
{
    // At least on Win2k server the chain is found to have dependencies on the original cert context.
    // So it should be closed first.
    if (chain != null)
    {
        chain.Reset();
    }
    if (remoteCertificateEx != null)
    {
        remoteCertificateEx.Reset();
    }
}
```